### PR TITLE
Filter parameters in iFrame url

### PR DESCRIPTION
**Pull request by khushbu-maurya**

### CODE REVIEW

**Comment by admin-vizcentric** [comment]
There were no errors, go have a cup of coffee...

File: src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js [hunk 1]
```diff
@@ -176,6 +176,19 @@
...
+                return item.featureType === "DIMENSION";
+            });
+
+            if (configuration.readOnly) {
+                var vms = states.viewState.visualMetadataSet || [];
+            } else {
+                var vms = states.visualMetadataSet || [];
+            }
+            vm.iFrames = vms.filter(function(item){
+                return item.metadataVisual.name === "Iframe";
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/flair-bi-content-header.controller.js:189]: Add iframe in constant/separate js file

File: src/main/webapp/app/entities/flair-bi/filter/filter-parameters.service.js [hunk 2]
```diff
@@ -435,5 +438,85 @@
                 valueType: 'dateRangeValueType'
             };
         }
+
+        function setFilterInIframeURL(filters,iframes,dashboardDatasource,dimensions) {
+
+            var filtersList = Object.keys(filters);
+            if (filtersList.length > 0) {
+                var datasourceId = dashboardDatasource;
+                iframes.forEach(element => {
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/filter/filter-parameters.service.js:447]: This is big function can you devide them into small functions

File: src/main/webapp/app/entities/flair-bi/filter/filter-parameters.service.js [hunk 3]
```diff
@@ -435,5 +438,85 @@
...
+                    var id = getParameterByName('datasourceId', element.properties[0].value)
+                    element.properties[0].value = removeURLParameter(element.properties[0].value, "filters");
+                    var filterUrl = {};
+                    if (datasourceId === id) {
+
+                        filtersList.forEach(item => {
```
Comment (johibkhan2) [inline src/main/webapp/app/entities/flair-bi/filter/filter-parameters.service.js:453]: Can we implement it without using more than one inner loops

**Pull request merged**